aptos-move/aptos-aggregator/src/aggregator_change_set.rs

```diff
@@ -2,331 +2,447 @@
-                delta,
-                history,
-                max_value,
-            } => merge_data_and_delta(base, delta, history, *max_value),
-            AggregatorChange::SnapshotDelta { delta, .. } => {
-                // History validation should already be done before this call.
-                expect_ok(BoundedMath::new(u128::MAX).unsigned_add_delta(base, delta))
-            },
+            AggregatorDelta { .. } => None,
+            SnapshotDelta {
+                base_aggregator, ..
+            } => Some(ApplyBase::Previous(*base_aggregator)),
+            SnapshotDerived { base_snapshot, .. } => Some(ApplyBase::Current(*base_snapshot)),
         }
     }
 
-    /// Applies self on top of previous change, merging them together. Note
-    /// that the strict ordering here is crucial for catching overflows correctly.
-    pub fn merge_with_previous_aggregator_change(
-        &mut self,
-        previous_change: &AggregatorChange,
-    ) -> PartialVMResult<()> {
-        match self {
-            // If the current state is Data, then merging with previous state won't change anything.
-            AggregatorChange::Data { .. } | AggregatorChange::StringData { .. } => Ok(()),
-            AggregatorChange::AggregatorDelta {
-                delta,
-                history,
-                max_value,
-            } => match previous_change {
-                AggregatorChange::Data { value: prev_value } => {
-                    let new_data = merge_data_and_delta(*prev_value, delta, history, *max_value)?;
-                    *self = AggregatorChange::Data { value: new_data };
-                    Ok(())
-                },
-                AggregatorChange::StringData { .. } => Err(code_invariant_error(
-                    "Base value cannot be string for an aggregator",
-                )),
-                AggregatorChange::AggregatorDelta {
-                    delta: prev_delta,
-                    max_value: prev_max_value,
-                    history: prev_history,
-                } => {
-                    if max_value != prev_max_value {
-                        return Err(code_invariant_error(
-                            "Max value of aggregator cannot change",
-                        ));
-                    }
-                    let (new_delta, new_history) =
-                        merge_two_deltas(prev_delta, prev_history, delta, history, *max_value)?;
-                    *self = AggregatorChange::AggregatorDelta {
-                        delta: new_delta,
-                        history: new_history,
-                        max_value: *max_value,
-                    };
-                    Ok(())
-                },
-                AggregatorChange::SnapshotDelta { .. } => Err(code_invariant_error(
-                    "SnapshotDelta cannot come before AggregatorDelta",
-                )),
+    pub fn get_apply_base_id(&self, self_id: &AggregatorID) -> ApplyBase {
```

Comment:
It's just 1 line code. Does this have to be a separate function?

Reply:
it is one line, but contains logic - i.e. ApplyBase::Previous(*self_id) - that is relevant for this file. 

and this gets called elsewhere , so encapsulating it here makes sense to me